sdk/src/main/scala/io/horizen/account/state/AccountState.scala

```diff
@@ -210,7 +210,11 @@ class AccountState(
       stateView.updateForgerBlockCounter(mod.forgerPublicKey, consensusEpochNumber)
 
       // If SC block has reached the end of the withdrawal epoch reward the forgers.
-      evalForgersReward(mod, modWithdrawalEpochInfo, consensusEpochNumber, stateView)
+      val distributionCap = if (Version1_4_0Fork.get(consensusEpochNumber).active) {
+        val mcLastBlockHeight = params.mainchainCreationBlockHeight + ((currentWithdrawalEpochInfo.epoch + 1) * params.withdrawalEpochLength) - 1
+        AccountFeePaymentsUtils.getMainchainWithdrawalEpochDistributionCap(mcLastBlockHeight, params.withdrawalEpochLength)
+      } else BigInteger.valueOf(Long.MaxValue)
+      evalForgersReward(mod, modWithdrawalEpochInfo, consensusEpochNumber, distributionCap, stateView)
```

Comment:
Actually `distributionCap` is required only in the end of withdrawal epoch. Wouldn't be better to calculate it inside the `evalForgersReward` method when this happens?

Reply:
Fixed, thanks!